### PR TITLE
feat: add `math/base/special/acscd`

### LINKED ISSUE #42: Implement `acscd`
Compute the inverse cosecant of `x` where the output is in degrees. `@stdlib/math/base/special/acscd`.

### PR DESCRIPTION
**Pull request by saisrikardumpeti**
Resolves #42 .

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds functionality to compute `acscd`  in degree of any given input.
-  Say we have input `x`, so it should satisfies this `x >= 1` or `x <= -1` else it will return `NaN` 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #42
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Pranavchiku** [comment]
@the-r3aper7 we appreciate your contribution but opening 10 PRs in a span of 10 hour just don't make any sense. Also, it unnecessarily blocks other people to work on those. Neither of PR seems perfect to me, even description is not filled correctly. Reiterating we prefer quality of contribution over number of PRs.

**Comment by saisrikardumpeti** [comment]
i am sorry, i will refactor them by today. i was just laying the foundation and I did not have intention to block other people work.

File: lib/node_modules/@stdlib/math/base/special/acscd/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+<section class="related">
+
+* * *
+
+## See Also
+
+-   <span class="package-name">[`@stdlib/math/base/special/acos`][@stdlib/math/base/special/acos]</span><span class="delimiter">: </span><span class="description">compute the arccosine of a double-precision floating-point number.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/asindh`][@stdlib/math/base/special/asindh]</span><span class="delimiter">: </span><span class="description">compute the hyperbolic arccosecant of a double-precision floating-point number.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/atan`][@stdlib/math/base/special/atan]</span><span class="delimiter">: </span><span class="description">compute the arccosecant of a double-precision floating-point number.</span>
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/README.md]: This section is auto-populated, can remove:

```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/acscd/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+[arccosecant]: https://en.wikipedia.org/wiki/Inverse_trigonometric_functions
+
+<!-- <related-links> -->
+
+[@stdlib/math/base/special/acos]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/acos
+
+[@stdlib/math/base/special/asindh]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/asindh
+
+[@stdlib/math/base/special/atan]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/atan
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/README.md]: This related links are auto-populated, can remove:

```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/acscd/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+Computes the [arccosecant][arccosecant] (in degree) of a double-precision floating-point number.
+
+```javascript
+var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var v = acscd( Infinity );
+// returns 0.0
+
+v = acscd( 2 * sqrt(3) / 3 );
+// returns 60.00000000000001
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/README.md]: We usually round to up to three digits after the decimal point and use `~` for approximate values in return annotations:

```suggestion
// returns ~60.0
```

File: lib/node_modules/@stdlib/math/base/special/acscd/README.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+v = acscd( sqrt(2) );
+// returns 44.99999999999999
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/README.md]:
```suggestion
// returns ~45.0
```

File: lib/node_modules/@stdlib/math/base/special/acscd/README.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+```javascript
+var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var v = acscd( Infinity );
+// returns 0.0
+
+v = acscd( 2 * sqrt(3) / 3 );
+// returns 60.00000000000001
+
+v = acscd( sqrt(2) );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/README.md]:
```suggestion
v = acscd( sqrt( 2 ) );
```

File: lib/node_modules/@stdlib/math/base/special/acscd/README.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+// returns 44.99999999999999
+
+v = acscd( 2 );
+// returns 30.000000000000004
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/README.md]:
```suggestion
// returns ~30.0
```

File: lib/node_modules/@stdlib/math/base/special/acscd/docs/repl.txt [hunk 7]
```diff
@@ -0,0 +1,30 @@
+
+{{alias}}( x )
+    Computes the arccosecant of a double-precision floating-point number.
+
+    If input does not satisfies `x >= 1` or `x <= -1` the function returns NaN.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/docs/repl.txt]:
```suggestion
    If `x` does not satisfy `x >= 1` or `x <= -1`, the function returns NaN.
```

File: lib/node_modules/@stdlib/math/base/special/acscd/lib/main.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+* @example
+* var v = acscd( NaN );
+* // returns NaN
+*/
+function acscd( x ) {
+	var rad;
+
+	rad = acsc(x);
+
+	return rad2deg(rad);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/lib/main.js]: When all variable declarations can be immediately assigned values, can collapse:

```suggestion
	var rad = acsc( x );
	return rad2deg( rad );
```

(notice also the added spaces around the function parentheses)

File: lib/node_modules/@stdlib/math/base/special/acscd/package.json [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+  "gypfile": true,
+  "directories": {
+    "benchmark": "./benchmark",
+    "doc": "./docs",
+    "example": "./examples",
+    "include": "./include",
+    "lib": "./lib",
+    "scripts": "./scripts",
+    "src": "./src",
+    "test": "./test"
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/package.json]: Directories need to be fixed here and potentially in the other PRs:

```suggestion
    "benchmark": "./benchmark",
    "doc": "./docs",
    "example": "./examples",
    "lib": "./lib",
    "test": "./test"
```

**Review by Planeshifter — COMMENTED** [review]
@the-r3aper7 Left a few comments here upon cursory look through the package. 

One important thing that should be added (here and in your other PRs) are tests showing that the function returns the correct values for valid inputs. You can generate a set of test values and add a test with description `the function computes the arccosecant (in degrees)` to `test.js`. Please have a look at other packages on how these are typically authored.

**Comment by saisrikardumpeti** [comment]
cool i will make changes to other PRs too. About tests part i have intentionally left those because I wanted to ask how can I generate that huge json file which have inputs.

**Comment by saisrikardumpeti** [comment]
@Planeshifter i have made those changes!

**Comment by saisrikardumpeti** [comment]
@Planeshifter i have made changes all the other PRs too. can you review those.

File: lib/node_modules/@stdlib/math/base/special/acscd/docs/types/index.d.ts [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/**
+* Computes the arccosecant (in degree) of a double-precision floating-point number.
+*
+* @param x - input value
+* @returns arcocsecant (in degree)
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/docs/types/index.d.ts]:
```suggestion
* @returns arccosecant (in degrees)
```

File: lib/node_modules/@stdlib/math/base/special/acscd/lib/main.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var rad2deg = require('@stdlib/math/base/special/rad2deg');
+var acsc = require('@stdlib/math/base/special/acsc');
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/lib/main.js]: Missing spaces around `require` parentheses:

```suggestion
var rad2deg = require( '@stdlib/math/base/special/rad2deg' );
var acsc = require( '@stdlib/math/base/special/acsc' );
```

File: lib/node_modules/@stdlib/math/base/special/acscd/package.json [hunk 12]
```diff
@@ -0,0 +1,70 @@
+{
+  "name": "@stdlib/math/base/special/acscd",
+  "version": "0.0.0",
+  "description": "Compute the arccosecant(degree) of a double-precision floating-point number.",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/package.json]:
```suggestion
  "description": "Compute the arccosecant (in degrees) of a double-precision floating-point number.",
```

File: lib/node_modules/@stdlib/math/base/special/acscd/package.json [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    "degree",
+    "arccosecant",
+    "sine",
+    "inverse",
+    "trig",
+    "trigonometry",
+    "radians",
+    "angle",
+    "polyfill",
+    "ponyfill"
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/package.json]:
```suggestion
    "angle"
```

File: lib/node_modules/@stdlib/math/base/special/acscd/lib/index.js [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Compute the arccosecant of a double-precision floating-point number.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/lib/index.js]:
```suggestion
* Compute the arccosecant (in degrees) of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/acscd/README.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# acscd
+
+> Compute the [arccosecant][arccosecant] in degree of a double-precision floating-point number.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/README.md]:
```suggestion
> Compute the [arccosecant][arccosecant] (in degrees) of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/acscd/README.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+
+## Usage
+
+```javascript
+var acscd = require( '@stdlib/math/base/special/acscd' );
+```
+
+#### acscd( x )
+
+Computes the [arccosecant][arccosecant] (in degree) of a double-precision floating-point number.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/README.md]:
```suggestion
Computes the [arccosecant][arccosecant] (in degrees) of a double-precision floating-point number.
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
This should be almost ready to merge now. Thanks for updating!

File: lib/node_modules/@stdlib/math/base/special/acscd/docs/repl.txt [hunk 17]
```diff
@@ -0,0 +1,30 @@
+
+{{alias}}( x )
+    Computes the arccosecant of a double-precision floating-point number.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/docs/repl.txt]:
```suggestion
    Computes the arccosecant (in degrees) of a double-precision floating-point
    number.
```

File: lib/node_modules/@stdlib/math/base/special/acscd/docs/repl.txt [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+
+    Parameters
+    ----------
+    x: number
+        Input value.
+
+    Returns
+    -------
+    y: number
+        Arccosecant (in degree).
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/docs/repl.txt]:
```suggestion
        Arccosecant (in degrees).
```

File: lib/node_modules/@stdlib/math/base/special/acscd/lib/main.js [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+
+
+// MAIN //
+
+/**
+* Computes the arccosecant (in degree) of a double-precision floating-point number.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/lib/main.js]:
```suggestion
* Computes the arccosecant (in degrees) of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/acscd/lib/main.js [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+*
+* @param {number} x - input value
+* @returns {number} arccosecant (in degree)
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/lib/main.js]:
```suggestion
* @returns {number} arccosecant (in degrees)
```

File: lib/node_modules/@stdlib/math/base/special/acscd/docs/types/index.d.ts [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/**
+* Computes the arccosecant (in degree) of a double-precision floating-point number.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/acscd/docs/types/index.d.ts]:
```suggestion
* Computes the arccosecant (in degrees) of a double-precision floating-point number.
```

**Comment by saisrikardumpeti** [comment]
cool i will make changes to other PRs. too

**Review by Planeshifter — APPROVED** [review]
Did one more look over the package; looks all good to me!

Will merge shortly.

**Pull request merged**